atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/DelegatingCassandraKeyValueServiceConfig.java

```diff
@@ -15,259 +15,231 @@
  */
 package com.palantir.atlasdb.cassandra;
 
-import com.google.common.base.MoreObjects;
+import com.palantir.atlasdb.cassandra.CassandraServersConfigs.CassandraServersConfig;
 import com.palantir.atlasdb.keyvalue.cassandra.async.CassandraAsyncKeyValueServiceFactory;
 import com.palantir.atlasdb.keyvalue.cassandra.pool.HostLocation;
-import com.palantir.atlasdb.spi.KeyValueServiceRuntimeConfig;
 import com.palantir.conjure.java.api.config.ssl.SslConfiguration;
 import java.net.InetSocketAddress;
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ExecutorService;
-import java.util.function.Function;
 import java.util.function.Supplier;
 
-public class CassandraReloadableKvsConfig implements CassandraKeyValueServiceConfig {
-    private final CassandraKeyValueServiceConfig config;
-    private final Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfigSupplier;
+public final class DelegatingCassandraKeyValueServiceConfig implements CassandraKeyValueServiceConfig {
 
-    public CassandraReloadableKvsConfig(
-            CassandraKeyValueServiceConfig config, Supplier<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
-        this.config = config;
-        this.runtimeConfigSupplier = runtimeConfig;
+    private final Supplier<CassandraKeyValueServiceConfig> delegate;
+
+    public DelegatingCassandraKeyValueServiceConfig(Supplier<CassandraKeyValueServiceConfig> delegate) {
```

Comment:
Typical delegating patterns (at least in AtlasDB) are an abstract class with a `delegate` method

Reply:
like what? all I'm seeing is similar to `DelegatingManagedTimestampService` which isn't abstract

Reply:
We should just use `@AutoDelegate` to generate the delegate implementation for us.

Reply:
`@AutoDelegate` requires us to put `@DoDelegate` on all 30 methods

Looks like atlasdb calls classes "Forwarding*" and yes uses `abstract delegate()`